python/lsst/pipe/analysis/colorAnalysis.py

```diff
@@ -74,28 +120,59 @@ def fromValues(cls, description, subDescription, plot, coeffs, requireGreater={}
 
 class NumStarLabeller(object):
     labels = {"star": 0, "maybe": 1, "notStar": 2}
-    plot = ["star"]
+    plot = ["star", "maybe"]
+
     def __init__(self, numBands):
         self.numBands = numBands
+
     def __call__(self, catalog):
-        return np.array([0 if nn == self.numBands else 2 if nn == 0 else 1 for nn in catalog["numStarFlags"]])
+        return np.array([0 if nn >= self.numBands else 2 if nn == 0 else 1 for nn in catalog["numStarFlags"]])
+
+
+class ColorValueInFitRange(object):
+    """Functor to produce color value if in the appropriate range
+
+    Here the range is set by upper and lower lines roughly perpendicular to the
+    fit where those lines cross the fit.  These numbers were previously determined
+    and are (currently) hard-wired in the fitLineSlope, fitLineUpperIncpt, and
+    fitLineLowerIncpt parameters in the ivezicTransformsHSC dict.
+    """
+    def __init__(self, column, xColor, yColor, fitLineSlope=None, fitLineUpperIncpt=None,
+                 fitLineLowerIncpt=None, unitScale=1.0):
+        self.column = column
+        self.xColor = xColor
+        self.yColor = yColor
+        self.fitLineSlope = fitLineSlope
+        self.fitLineUpperIncpt = fitLineUpperIncpt
+        self.fitLineLowerIncpt = fitLineLowerIncpt
+        self.unitScale = unitScale
+
+    def __call__(self, principleColCats):
+        good = np.ones(len(principleColCats), dtype=bool)
```

Comment:
Probably fine without this line and changing the following `&=` into `=`:
`good &= self.yColor > self.fitLineLowerIncpt + self.fitLineSlope*self.xColor` I'm assuming principleColCats[self.column] and self.yColor are the same shape?